src/Cli/dotnet/commands/dotnet-tool/install/ToolInstallLocalCommand.cs

```diff
@@ -46,16 +56,25 @@ internal class ToolInstallLocalCommand : CommandBase
                                   new ToolManifestFinder(new DirectoryPath(Directory.GetCurrentDirectory()));
             _toolManifestEditor = toolManifestEditor ?? new ToolManifestEditor();
             _localToolsResolverCache = localToolsResolverCache ?? new LocalToolsResolverCache();
-            _toolLocalPackageInstaller = new ToolInstallLocalInstaller(parseResult, toolPackageDownloader);
+            _toolLocalPackageInstaller = new ToolInstallLocalInstaller(parseResult, _packageId, toolPackageDownloader);
+            _toolPackageDownloader = toolPackageDownloader;
             _allowRollForward = parseResult.GetValue(ToolInstallCommandParser.RollForwardOption);
-
-            _packageId = new PackageId(parseResult.GetValue(ToolUpdateCommandParser.PackageIdArgument));
             _allowPackageDowngrade = parseResult.GetValue(ToolInstallCommandParser.AllowPackageDowngradeOption);
-
         }
-        
 
         public override int Execute()
+        {
+            if (_all)
+            {
+                return ExecuteInstallAllCommand();
```

Comment:
```suggestion
                 // return ExecuteInstallAllCommand();
                var toolListCommand = new ToolListLocalCommand(_parseResult, (IToolManifestInspector)_toolManifestFinder);
                var toolIds = toolListCommand.GetPackages(null);
                foreach (var toolId in toolIds)
                {
                    ExecuteInstallCommand(new PackageId(toolId.Item1.PackageId.ToString().ToString()));
                }
                return 0;
```

Thanks to @dsplaisted suggestions, we can probably pass `packageId` to the `ExecuteInstallCommand` so that we don't need to build an install command from scratch

Reply:
Done!